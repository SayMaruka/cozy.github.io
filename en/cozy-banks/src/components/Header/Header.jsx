import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import styles from 'components/Header/Header.styl'
import CozyTheme from 'cozy-ui/transpiled/react/CozyTheme'

class Header extends React.PureComponent {
  render() {
    const { children, className, theme, fixed } = this.props

    return (
      <CozyTheme variant={theme}>
        <div
          className={cx(
            styles[`HeaderColor_${theme}`],
            { [styles.HeaderFixed]: fixed },
            className
          )}
        >
          {children}
        </div>
      </CozyTheme>
    )
  }
}

Header.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  theme: PropTypes.oneOf(['normal', 'inverted'])
}

Header.defaultProps = {
  theme: 'normal',
  fixed: false
}

export default Header
