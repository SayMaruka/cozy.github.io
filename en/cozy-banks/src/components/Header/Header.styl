@require 'settings/breakpoints'

.HeaderFixed
    position sticky
    position -webkit-sticky
    z-index 2
    top 0

    +medium-screen()
        top 3rem

.HeaderColor_default
    background-color var(--headerDefaultBackgroundColor)

.HeaderColor_primary
    background-color var(--headerPrimaryBackgroundColor)

.HeaderColor_primary.HeaderFixed
    box-shadow 0 2px 4px 0 rgba(0, 0, 0, 0.24), 0 0 4px 0 rgba(0, 0, 0, 0.16)
