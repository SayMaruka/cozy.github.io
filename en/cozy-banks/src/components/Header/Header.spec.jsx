import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('Header', () => {
  it(`should display children`, () => {
    expect(shallow(<Header>content</Header>).getElement()).toMatchSnapshot()
  })

  it(`should extend className`, () => {
    expect(
      shallow(<Header className="noPaddingBottom">content</Header>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set position fixed`, () => {
    expect(
      shallow(<Header fixed>content</Header>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set color default`, () => {
    expect(
      shallow(<Header color="default">content</Header>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set color primary`, () => {
    expect(
      shallow(<Header color="primary">content</Header>).getElement()
    ).toMatchSnapshot()
  })
})
