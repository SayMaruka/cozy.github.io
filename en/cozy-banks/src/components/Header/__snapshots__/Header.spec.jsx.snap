// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should display children 1`] = `
<div
  className="HeaderColor_default"
>
  content
</div>
`;

exports[`Header should extend className 1`] = `
<div
  className="HeaderColor_default noPaddingBottom"
>
  content
</div>
`;

exports[`Header should set color default 1`] = `
<div
  className="HeaderColor_default"
>
  content
</div>
`;

exports[`Header should set color primary 1`] = `
<div
  className="HeaderColor_primary"
>
  content
</div>
`;

exports[`Header should set position fixed 1`] = `
<div
  className="HeaderColor_default HeaderFixed"
>
  content
</div>
`;
